### ISSUE TITLE
Enabling "Comments" attribute and passing additional validation results

### ISSUE THREAD

**Issue by justinasr** [dmwm/WMCore#9737]
**Impact of the new feature**
ReqMgr2

**Is your feature request related to a problem? Please describe.**
Not related to a problem.

**Describe the solution you'd like**
PdmV would like to pass additional (optional) information about request - CPU efficiency and PeakValueRSS values obtained from running "validation" in HTCondor with 1, 2, 4 and 8 cores. We were discussing possibility to pass this as "Comments" attribute. As we discussed, "Comments" attribute are not yet available in production environment, so it would have to be *enabled* too. Initial idea was to pass it as:
```
values = [{
  "cores":1
  "peak_value_rss": 422.461,
  "cpu_efficiency": 0.9975137405508916
 },
 {
  "cores":2,
  "peak_value_rss": 425.336,
  "cpu_efficiency": 0.8232267128434672
 },
 {
  "cores":4,
  "peak_value_rss": 451.078,
  "cpu_efficiency": 0.5283693977716476
 },
 {
  "cores": 8,
  "peak_value_rss": 473.219,
  "cpu_efficiency": 0.29084418625506936
}]
...
reqmgr2_dict["Comments"] = json.dumps(values)
...
```
However, I don't like the idea of "sacrificing" "Comments" to only these validation values. I would like to propose to wrap these values in another dictionary:
```
reqmgr2_dict["Comments"] = json.dumps({"validation": values})
```
In case someone would like to pass more values in "Comments", such as "pilot" flags (related issue: https://github.com/CMSCompOps/WmAgentScripts/issues/585):
```
reqmgr2_dict["Comments"] = json.dumps({"validation": values, "pilot": True, "something_else": ...})
```
This would be a very small in JSON string length and would have a minimal impact on total database size.
Of course, at the moment, main goal is to pass validation values, but investing 15 more bytes now to make this more flexible and future-proof is also worth considering.

**Describe alternatives you've considered**
None.

**Additional context**
Values that we have:
```
{
 "cores": 1,
 "peak_value_rss": 422.461,
 "total_events": 10000,
 "time_per_event": 0.18819774313266435,
 "filter_efficiency": 1,
 "cpu_efficiency": 0.9975137405508916,
 "size_per_event": 101.0944
}
```
As agreed in email thread, only cores, peak_value_rss and cpu_efficiency will be passed.

**Comment by wajidalikhan** [comment]
Thanks a lot Justinus!

**Comment by haozturk** [comment]
Hi @justinasr just to confirm what we've just discussed with Jordan: do these numbers only belong to the task1? i.e. there is no info for Task2 and further, because you do not run validation for them, right?

**Comment by justinasr** [comment]
> Hi @justinasr just to confirm what we've just discussed with Jordan: do these numbers only belong to the task1? i.e. there is no info for Task2 and further, because you do not run validation for them, right?

Hello, usually yes, but there are cases, like pLHEGEN where both Task1 (pLHE) and Task2 (GEN) are validated and could have their own "Comments". At the moment both tasks share the same values from validation, but this should be improved in our side.

**Comment by justinasr** [comment]
Hello @haozturk, sure, in production or testbed? Is ReqMgr2 ready to accept "Comments"?

**Comment by haozturk** [comment]
> Hello @haozturk, sure, in production or testbed? 

Production should be fine, but let's do not switch it for all requests for now. If you can submit ~10 requests with comments, that should be okay to do the first tests.

> Is ReqMgr2 ready to accept "Comments"?

I assume ReqMgr2 does not do any validation against "Comments", but let's wait for the confirmation of @amaltaro  or @todor-ivanov  for that.

**Comment by amaltaro** [comment]
Honestly speaking, I would rather see this information being consumed by McM and McM itself taking the decision whether a TaskChain or a StepChain workflow is going to be created.

Nope, there is not validation *yet* on this field, but as we start using it, we should definitely implement something to make sure input data is "under control".

**Comment by haozturk** [comment]
I agree that killing&cloning a taskchain and creating a new stepchain workflow is not the cleanest way. But we're still working towards implementing a better conversion algorithm [1,2,3,4,5]. Migrating the code from one place to another would add another level of complication in this improvement and slow down the progress. So, I'd suggest continuing what we have been doing so far until we make sure we have an algorithm close to optimal. Then we can discuss migration of the code, IMO.

Doing some tests on this information will take some time, if it's decided to use this information in production, then we can also start adding a control for this field.

@justinasr please let me know when you inject the requests. 

[1] https://github.com/CMSCompOps/WmAgentScripts/issues/837 
[2] https://github.com/CMSCompOps/WmAgentScripts/issues/827
[3] https://github.com/CMSCompOps/WmAgentScripts/issues/829
[4] https://github.com/CMSCompOps/WmAgentScripts/issues/831
[5] https://github.com/CMSCompOps/WmAgentScripts/issues/824
